java/main/src/test/java/com/saucelabs/saucebindings/integration/PerformanceTest.java

```diff
@@ -0,0 +1,168 @@
+package com.saucelabs.saucebindings.integration;
+
+import com.saucelabs.saucebindings.SauceSession;
+import com.saucelabs.saucebindings.options.SauceOptions;
+import com.saucelabs.saucebindings.performance.JankinessResults;
+import com.saucelabs.saucebindings.performance.PerformanceMetrics;
+import com.saucelabs.saucebindings.performance.PerformanceResults;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TestName;
+import org.openqa.selenium.By;
+import org.openqa.selenium.remote.RemoteWebDriver;
+import org.yaml.snakeyaml.Yaml;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+public class PerformanceTest {
+    private SauceSession session;
+    private RemoteWebDriver driver;
+
+    @After
+    public void cleanUp() {
+        if (session != null) {
+            session.stop(true);
+        }
+    }
+
+    @Rule
+    public TestName testName = new TestName();
+
+    @Before
+    public void setup() {
+        SauceOptions sauceOptions = SauceOptions.chrome()
+                .setName(testName.getMethodName())
+                .setCapturePerformance()
+                .build();
+        session = new SauceSession(sauceOptions);
+        driver = session.start();
+    }
+
+    @Test
+    public void jankinessResults() {
+        driver.get("https://www.saucedemo.com");
+
+        JankinessResults jankinessResults = session.performance().getJankinessResults();
+        Assert.assertTrue(jankinessResults.getScore() > 0.7);
+    }
+
+    @Test
+    public void performanceResults() {
+        driver.get("https://www.saucedemo.com");
+
+        PerformanceResults performanceResults = session.performance().getResults();
+        // We don't actually care if this "passes" or "fails" just that these methods are returning correct values
+        if (performanceResults.isPassed()) {
+            Assert.assertTrue(performanceResults.getDetails().isEmpty());
+            Assert.assertNull(performanceResults.getReason());
+        } else {
+            Assert.assertTrue(performanceResults.getReason().contains("Metrics"));
+            Assert.assertFalse(performanceResults.getDetails().isEmpty());
+        }
+    }
+
+    @Test
+    public void performanceMetricResults() {
+        driver.get("https://www.saucedemo.com");
+
+        PerformanceResults performanceResults = session.performance().getResults("firstInteractive");
```

Comment:
@titusfortner should we move the parameter values into an enum? They've remained the same forever and it should improve usability

Reply:
Yeah, so long as we have `getRawResults()` I don't think there's a problem doing enums for this.